packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -1782,16 +1803,18 @@ const double _maxCalendarWidthPortrait = 480.0;
 
 /// Displays a scrollable calendar grid that allows a user to select a range
 /// of dates.
-class _CalendarDateRangePicker extends StatefulWidget {
+class CalendarDateRangePicker extends StatefulWidget {
```

Comment:
Why is this made public?

Reply:
To do the same as `CalendarDatePicker` that is already public, and also to allow users to display the calendar picker without a bottom sheet drawer. It's highly useful for non-mobile or large screens.

Reply:
Isn't that out of scope of this PR ? Is this required by this feature ? If not I suggest opening a separate one, as it's unlikely both unrelated changes will be merged together

Reply:
I've been going back and forward on whether to just let this be included, but let's keep this as private for this PR. If we make it public, it would need a pass over it's documentation and probably a test for how it behaves outside of a `DateRangePickerDialog`, which should be it's own PR if it's not directly needed.

We're planning to take a deeper look into our calendar widgets later in the year, and this can wait for that.

Reply:
Later means when? But yes I'll make it back to private, and tackle it inside another PR.

Reply:
```suggestion
class _CalendarDateRangePicker extends StatefulWidget {
```

Reply:
@Chuckame if you implement another PR for making this public or the customization we talked about in another thread please ping me.

 > Later means when
 
 I doubt any time frame should be relied upon here.